src/main/java/com/myhome/services/springdatajpa/CommunitySDJpaService.java

```diff
@@ -66,59 +68,49 @@ public CommunitySDJpaService(CommunityRepository communityRepository,
   }
 
   @Override public Optional<Community> getCommunityDetailsById(String communityId) {
-    Community community = communityRepository.findByCommunityId(communityId);
-    return community == null ? Optional.empty() : Optional.of(community);
+    return communityRepository.findByCommunityId(communityId);
   }
 
-  @Override public Community addAdminsToCommunity(String communityId, Set<String> admins) {
-    if (!getCommunityDetailsById(communityId).isPresent()) return new Community();
-    Community community = communityRepository.findByCommunityId(communityId);
-
-    Set<CommunityAdmin> savedAdminSet = new HashSet<>();
-    admins.forEach(s -> {
-      CommunityAdmin admin = new CommunityAdmin();
-      admin.setAdminId(s);
-      admin.getCommunities().add(community);
-      savedAdminSet.add(communityAdminRepository.save(admin));
-    });
-    community.getAdmins().addAll(savedAdminSet);
-    return communityRepository.save(community);
+  @Override public Community addAdminsToCommunity(String communityId, Set<String> adminsIds) {
+    Optional<Community> communitySearch = communityRepository.findByCommunityId(communityId);
+    return communitySearch.map(community -> {
+      adminsIds.forEach(adminId -> {
+        communityAdminRepository.findByAdminId(adminId).map(admin -> {
+          admin.getCommunities().add(community);
+          community.getAdmins().add(communityAdminRepository.save(admin));
+          return admin;
+        });
+      });
+      return communityRepository.save(community);
+    }).orElseGet(() -> new Community());
   }
 
   // Returns houseId which was added to the community
   @Override
   public Set<String> addHousesToCommunity(String communityId, Set<CommunityHouse> houses) {
-    if (!getCommunityDetailsById(communityId).isPresent()) return new HashSet<>();
-    Community community = communityRepository.findByCommunityId(communityId);
-    houses.stream().map(house -> {
-      house.setHouseId(generateUniqueId());
-      return house;
-    }).forEach(communityHouse -> communityHouse.setCommunity(community));
-    Set<CommunityHouse> savedHouses = new HashSet<>();
-    communityHouseRepository.saveAll(houses).forEach(savedHouses::add);
-    community.getHouses().addAll(savedHouses);
-    communityRepository.save(community);
-
-    Set<String> houseIds = new HashSet<>(savedHouses.size());
-    savedHouses.forEach(communityHouse -> houseIds.add(communityHouse.getHouseId()));
-    return houseIds;
+    Optional<Community> communitySearch = communityRepository.findByCommunityId(communityId);
+    return communitySearch.map(community -> {
+      Set<CommunityHouse> savedHouses = houses.stream()
+          .filter(house -> communityHouseRepository.findByHouseId(house.getHouseId()) != null)
+          .peek(house -> house.setCommunity(community))
```

Comment:
Prefer map over peek. Peek is more for debugging. 

https://stackoverflow.com/questions/44370676/java-8-peek-vs-map

Reply:
Yes, let's create new object to have immutability here.
Lombok has @With or @Wither annotation to simplify this